backend/logic/courses/courses.js

```diff
@@ -0,0 +1,10 @@
+const mysql = require('../../sql/connection');
+
+async function queryCourseByTag(tag){
+    //if an invalid tag is passed, an empty object is returned
+    let connection = await mysql.getNewConnection();
+    let courses = await connection.query('SELECT course_code FROM course_tag WHERE course_tag.tag_name LIKE ?', tag);
+    return (JSON.parse(JSON.stringify(courses)));
```

Comment:
The connection needs to be released before you return courses

Reply:
Modified, thanks!